tests/cli/conftest.py

```diff
@@ -39,16 +38,9 @@
 )
 
 
-@pytest.fixture(scope='function', autouse=True)
-def stdout_trap(mocker):
-    trap = StringIO()
-    mocker.patch('sys.stdout', new=trap)
-    return trap
-
-
 @pytest.fixture(scope='function')
-def test_emitter(mocker, stdout_trap):
-    mocker.patch('sys.stdout', new=stdout_trap)
+def test_emitter(mocker):
+    # Stdout will go either to `capsys` or to the console depending on the run settings.
```

Comment:
:+1:  Really not obvious... This needs to be announced in big bold letters somewhere easy for contributors to find.

Reply:
Yes, I wasn't sure how to handle it. The only consolation is that is wasn't obvious before as well... What's the expected behavior here? If you ask for a `test_emitter` fixture, where would you expect the output to end up?

Reply:
In fact, I think the current variant follows the principle of least surprise. If you ask for `test_emitter`, you get an emitter object to give to functions that require it. If you ask for `capsys`, you get the output captured. If you ask for both, you get both.

Reply:
Yeah, makes sense :+1:

Reply:
Anyway, I added a slightly more descriptive comment.